authentication/auth.go

```diff
@@ -119,6 +120,66 @@ type Metadata struct {
 	Default   string `json:"default"`
 }
 
+func isSurveyMetadata(key string) bool {
+	switch key {
+	case
+		"case_ref",
+		"case_type",
+		"display_address",
+		"employment_date",
+		"form_type",
+		"period_id",
+		"period_str",
+		"ref_p_end_date",
+		"ref_p_start_date",
+		"ru_name",
+		"ru_ref",
+		"trad_as",
+		"user_id",
+		"questionnaire_id":
+		return true
+	}
+	return false
+}
+
+func isRequiredMetadata(key string) bool {
+	switch key {
+	case
+		"case_id",
+		"region_code",
+		"channel",
+		"collection_exercise_sid",
+		"response_expires_at",
+		"response_id",
+		"schema_name",
+		"account_service_url":
+		return true
+	}
+	return false
+}
+
+func getSurveyMetadataFromClaims(claimValues map[string][]string, data map[string]interface{}, claims map[string]interface{}, surveyMetadata map[string]interface{}) {
+	for key, value := range claimValues {
+		if isSurveyMetadata(key) == true {
```

Comment:
Here you can make it shorter as you did on line 167
```suggestion
		if isSurveyMetadata(key) {
```